bot/exts/evergreen/githubinfo.py

```diff
@@ -0,0 +1,88 @@
+import logging
+from datetime import datetime
+
+import aiohttp
+import discord
+from discord.ext import commands
+from discord.ext.commands.cooldowns import BucketType
+
+log = logging.getLogger(__name__)
+
+
+class GithubInfo(commands.Cog):
+    """Fetches GitHub."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    async def fetch_data(self, url: str) -> dict:
+        """Retrieve data as a dictionary."""
+        async with aiohttp.ClientSession() as cs:
+            async with cs.get(url) as r:
+                return await r.json()
+
+    @commands.command(name='github', aliases=['gh'])
+    @commands.cooldown(1, 5, BucketType.user)
+    async def get_github_info(self, ctx: commands.Context, username: str = 'python-discord') -> None:
+        """
+        Fetches a user's GitHub information.
+
+        User defaults to python-discord.
+
+        Usage: .gh [username]
+        """
+        # fetch user data
```

Comment:
While everything from here to line 82 (the line before `await ctx.send(embed=embed)`) is happening, it would be optimal if it looks like the bot is typing. Like this:
```
async with ctx.typing():
    user_data = await self.fetch_data(f"https://api.github.com/users/{username}")
    # ...
    embed.add_field(name="Blog", value=blog)

await ctx.send(embed=embed)
```
This (`async with ctx.typing():`) is using a context manager and everything between the lines I included above should be indented, to be a part of what happens while the bot is "typing".

Reply:
This might be redundant since those lines don't really take a long amount of time to execute and we might not see the bot typing for a time period large enough to notice, if at all. That said, this is definitely a cool addition and I'll add it anyways

Reply:
I think it's long enough to start wondering if the bot really saw the message. It's always nice to get some confirmation.